src/ralph/backends/data/clickhouse.py

```diff
@@ -0,0 +1,395 @@
+
+    where_clause: Optional[str]
+    return_fields: Optional[List[str]]
+
+
+class ClickHouseDataBackend(
+    BaseDataBackend
+):  # pylint: disable=too-many-instance-attributes
+    """ClickHouse database backend."""
+
+    name = "clickhouse"
+    query_model = ClickHouseQuery
+    default_operation_type = BaseOperationType.CREATE
+    settings_class = ClickHouseDataBackendSettings
+
+    def __init__(self, settings: settings_class = None):
+        """Instantiate the ClickHouse configuration.
+
+        Args:
+            settings (ClickHouseDataBackendSettings): The ClickHouse data backend
+                settings.
+        """
+        self.settings = settings if settings else self.settings_class()
+        self.database = self.settings.DATABASE
+        self.event_table_name = self.settings.EVENT_TABLE_NAME
+        self.default_chunk_size = self.settings.DEFAULT_CHUNK_SIZE
+        self.locale_encoding = self.settings.LOCALE_ENCODING
+        self._client = None
+
+    @property
+    def client(self):
+        """Create a ClickHouse client if it doesn't exist.
+
+        We do this here so that we don't interrupt initialization in the case
+        where ClickHouse is not running when Ralph starts up, which will cause
+        Ralph to hang. This client is HTTP, so not actually stateful. Ralph
+        should be able to gracefully deal with ClickHouse outages at all other
+        times.
+        """
+        if not self._client:
+            self._client = clickhouse_connect.get_client(
+                host=self.settings.HOST,
+                port=self.settings.PORT,
+                database=self.database,
+                username=self.settings.USERNAME,
+                password=self.settings.PASSWORD,
+                settings=self.settings.CLIENT_OPTIONS.dict(),
+            )
+        return self._client
+
+    def status(self) -> DataBackendStatus:
+        """Check ClickHouse connection status.
+
+        Returns:
+            DataBackendStatus: The status of the data backend.
+        """
+        try:
+            self.client.query("SELECT 1")
+        except ClickHouseError:
+            return DataBackendStatus.AWAY
+
+        return DataBackendStatus.OK
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        """List documents for a given table.
+
+        Args:
+            target (str): The table to list documents from.
+            details (bool): Get detailed archive information instead of just ids.
+            new (bool): Given the history, list only not already fetched archives.
+        """
+        sql = "SHOW TABLES"
+        self.client.database = target if target else self.database
+        for document in self.client.query(sql).named_results():
+            if details:
+                yield document
+            else:
+                yield str(document.get("name"))
+
+    @enforce_query_checks
+    def read(
```

Comment:
Could we add logging information  to indicate that read has started (and/or has been completed) ?